components/support/nimbus-fml/src/backends/kotlin/mod.rs

```diff
@@ -0,0 +1,162 @@
+        join(sdk_dir(), "android/src/main/java")
+    }
+
+    // The directory with the mock implementations of Android
+    // used for testing.
+    fn runtime_dir() -> String {
+        join(pkg_dir(), "fixtures/android/runtime")
+    }
+
+    // We'll put our test scripts in here.
+    fn tests_dir() -> String {
+        join(pkg_dir(), "fixtures/android/tests")
+    }
+
+    // The jar archive we need to do JSON with in Kotlin/Java.
+    // This is the same library as bundled in Android.
+    fn json_jar() -> String {
+        join(runtime_dir(), "json.jar")
+    }
+
+    // The file with the kt implementation of FeatureVariables
+    fn variables_kt() -> String {
+        join(
+            sdk_android_dir(),
+            "org/mozilla/experiments/nimbus/FeatureVariables.kt",
+        )
+    }
+
+    fn nimbus_internals_kt() -> String {
+        join(sdk_android_dir(), "org/mozilla/experiments/nimbus/internal")
+    }
+
+    // The file with the kt implementation of FeatureVariables
+    fn features_kt() -> String {
+        join(
+            sdk_android_dir(),
+            "org/mozilla/experiments/nimbus/FeaturesInterface.kt",
+        )
+    }
+
+    fn classpath(classes: &Path) -> Result<String> {
+        Ok(format!("{}:{}", json_jar(), classes.to_str().unwrap()))
+    }
+
+    fn detect_kotlinc() -> Result<bool> {
+        let output = Command::new("which").arg("kotlinc").output()?;
+
+        Ok(output.status.success())
+    }
+
+    // Compile a genertaed manifest file against the mocked out Android runtime.
+    pub fn compile_manifest_kt(manifest_path: String) -> Result<TempDir> {
+        let path = PathBuf::from(&manifest_path);
+        let prefix = path
+            .file_stem()
+            .ok_or_else(|| FMLError::InvalidPath(manifest_path.clone()))?;
+        let prefix = prefix
+            .to_str()
+            .ok_or_else(|| FMLError::InvalidPath(manifest_path.clone()))?;
+        let temp = TempDir::new(prefix)?;
+        let build_dir = temp.path();
+
+        let status = Command::new("kotlinc")
+            // Our generated bindings should not produce any warnings; fail tests if they do.
+            .arg("-Werror")
+            .arg("-J-ea")
+            // Reflect $CLASSPATH from the environment, to help find `json.jar`.
+            .arg("-classpath")
+            .arg(json_jar())
+            .arg("-d")
+            .arg(&build_dir)
+            .arg(&variables_kt())
+            .arg(&features_kt())
+            .arg(&runtime_dir())
+            .arg(&nimbus_internals_kt())
+            .arg(&manifest_path)
+            .spawn()?
+            .wait()?;
+        if status.success() {
+            Ok(temp)
+        } else {
+            bail!("running `kotlinc` failed compiling a generated manifest")
+        }
+    }
+
+    // Given a generated manifest, run a kts script against it.
+    pub fn run_script_with_generated_code(manifest_kt: String, script: &str) -> Result<()> {
+        if !detect_kotlinc()? {
```

Comment:
Nice! FWIW the `println!` won't be seen on `cargo test` since cargo test defaults to catching `stdout` but that's OK for now!